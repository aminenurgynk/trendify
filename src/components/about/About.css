.carousel-slide {
    position: relative;
    text-align: center;
    height: 525px; 
    background: #364d79;
    overflow: hidden;
  }
  
  .carousel-slide img {
    height: 575px; 
    object-fit: cover; 
  }
  
  .carousel-caption {
    position: absolute;
    bottom: 20%;
    left: 50%;
    transform: translateX(-50%);
    color: #ffffff;
    text-align: center;
    padding: 20px;
    background: rgba(0, 0, 0, 0.5); 
  }
  
  .carousel-caption h3 {
    font-size: 1.8em;
  }
  
  .carousel-caption p {
    font-size: 1em;
    margin-top: 0.5em;
  }
  