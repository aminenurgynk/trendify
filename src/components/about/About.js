import React from "react";
import { Carousel } from "antd";
import "./About.css";

const About = () => {
  return (
    <Carousel autoplay>
      <div className="carousel-slide">
        <img
          src="https://static.zarahome.net/8/contentEcom/dto_imagen/home/slider/desktop/Slider_playa.jpg?20240323021206&imformat=chrome&imwidth=1827&impolicy=zarahome-itxmedium"
          alt="Item 1"
          className="carousel-img1"
          width="100%"
        />
        <div className="carousel-caption">
          <h3>New Season Fashion</h3>
          <p>Discover the Spring / Summer '24 collection.</p>
        </div>
      </div>

      <div className="carousel-slide">
        <img
          src="https://static.pullandbear.net/2/cms/assets/uploads/hstwdsv_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
          alt="Item 1"
          className="carousel-img1"
          width="100%"
        />
        <div className="carousel-caption">
          <h3>New Season Fashion</h3>
          <p>Discover the Spring / Summer '24 collection.</p>
        </div>
      </div>

      <div className="carousel-slide">
        <img
          src="https://static.pullandbear.net/2/cms/assets/uploads/landscape29_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
          alt="Item 1"
          className="carousel-img1"
          width="100%"
        />
        <div className="carousel-caption">
          <h3>New Season Fashion</h3>
          <p>Discover the Spring / Summer '24 collection.</p>
        </div>
      </div>

      <div className="carousel-slide">
        <img
          src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
          alt="Item 2"
        />
        <div className="carousel-caption">
          <h3>New Season Fashion</h3>
          <p>Discover the Spring / Summer '24 collection.</p>
        </div>
      </div>
    </Carousel>
  );
};

export default About;
