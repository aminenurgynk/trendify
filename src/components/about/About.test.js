import { render, screen } from '@testing-library/react';
import About from './About'; // Update path if necessary

describe('About component', () => {
  it('should render the carousel with multiple slides', () => {
    render(<About />);
    const carousel = screen.getByRole('carousel'); // Look for element with role="carousel"
    expect(carousel).toBeInTheDocument();
  });

  it('should render the title "New Season Fashion" within slides', () => {
    render(<About />);
    const titles = screen.getAllByText(/New Season Fashion/i); // Look for all elements containing "New Season Fashion" (case-insensitive)
    expect(titles.length).toBeGreaterThanOrEqual(1); // Expect at least one title
  });

  // Add more tests for specific functionalities if needed
});

it('should render the About component correctly', () => {
  const { asFragment } = render(<About />);
  expect(asFragment()).toMatchSnapshot();
});
