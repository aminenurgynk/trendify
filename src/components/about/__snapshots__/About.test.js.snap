// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-carousel css-dev-only-do-not-override-djtmh8"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.zarahome.net/8/contentEcom/dto_imagen/home/slider/desktop/Slider_playa.jpg?20240323021206&imformat=chrome&imwidth=1827&impolicy=zarahome-itxmedium"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/hstwdsv_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/landscape29_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="4"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.zarahome.net/8/contentEcom/dto_imagen/home/slider/desktop/Slider_playa.jpg?20240323021206&imformat=chrome&imwidth=1827&impolicy=zarahome-itxmedium"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/hstwdsv_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="6"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/landscape29_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="7"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots slick-dots-bottom"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render the About component correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-carousel css-dev-only-do-not-override-djtmh8"
  >
    <div
      class="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        class="slick-list"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="-1"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.zarahome.net/8/contentEcom/dto_imagen/home/slider/desktop/Slider_playa.jpg?20240323021206&imformat=chrome&imwidth=1827&impolicy=zarahome-itxmedium"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="1"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/hstwdsv_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="2"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/landscape29_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide"
            data-index="3"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="4"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.zarahome.net/8/contentEcom/dto_imagen/home/slider/desktop/Slider_playa.jpg?20240323021206&imformat=chrome&imwidth=1827&impolicy=zarahome-itxmedium"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="5"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/hstwdsv_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="6"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 1"
                  class="carousel-img1"
                  src="https://static.pullandbear.net/2/cms/assets/uploads/landscape29_1.jpg?imwidth=1500&impolicy=pullandbear-itxmediumhigh&imformat=chrome&ts=20240323020706"
                  width="100%"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="slick-slide slick-cloned"
            data-index="7"
            style="width: 0px;"
            tabindex="-1"
          >
            <div>
              <div
                class="carousel-slide"
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                <img
                  alt="Item 2"
                  src="https://st-levis.mncdn.com/Content/img/Unisite/web.jpg"
                />
                <div
                  class="carousel-caption"
                >
                  <h3>
                    New Season Fashion
                  </h3>
                  <p>
                    Discover the Spring / Summer '24 collection.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="slick-dots slick-dots-bottom"
        style="display: block;"
      >
        <li
          class="slick-active"
        >
          <button>
            1
          </button>
        </li>
        <li
          class=""
        >
          <button>
            2
          </button>
        </li>
        <li
          class=""
        >
          <button>
            3
          </button>
        </li>
        <li
          class=""
        >
          <button>
            4
          </button>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
